src/submodule.c

```diff
@@ -149,40 +149,53 @@ static int find_by_path(const git_config_entry *entry, void *payload)
 }
 
 /**
- * Find out the name of a submodule from its path
+ * Release the name map returned by 'load_submodule_names'.
  */
-static int name_from_path(git_buf *out, git_config *cfg, const char *path)
+static void free_submodule_names(git_strmap *names)
+{
+	git_buf *name = 0;
+	if (0 == names)
```

Comment:
Please don't reverse these tests.  I know that many people have gotten into a habit of this and find it a good practice, but please follow our existing code style.

Please don't use `0` for pointers.

This should be `if (names == NULL)`.